src/test/integration/suite/deployment-logs.test.it.ts

```diff
@@ -0,0 +1,151 @@
+let secondEnvironmentMock: EnvironmentModel;
+
+suite("deployment logs", function () {
+  this.timeout(1000 * 10);
+
+  beforeEach(async () => {
+    firstEnvironmentMock = getEnvironmentMock(
+      "main",
+      "https://github.com/user/repo",
+      {
+        name: firstEnvName,
+      }
+    );
+    secondEnvironmentMock = getEnvironmentMock(
+      "main",
+      "https://github.com/user/repo",
+      {
+        name: secondEnvName,
+        id: "second-env-id",
+        latestDeploymentLog: {
+          ...firstEnvironmentMock.latestDeploymentLog,
+          id: "second-deployment-id",
+        },
+      }
+    );
+    mockOutputChannel();
+    await resetExtension(); // we need to resat because we are mocking the output channel
+    const environments = [firstEnvironmentMock, secondEnvironmentMock];
+    mockGetOrganization(orgId, auth);
+    mockGetEnvironment(orgId, environments, auth);
+    mockGitRepoAndBranch("main", "git@github.com:user/repo.git");
+    await login(auth);
+    await waitFor(() => expect(getFirstEnvStatus()).toBe("ACTIVE"));
+  });
+
+  afterEach(async () => {
+    await logout();
+    await resetExtension();
+    resetOutputChannelMocks();
+  });
+
+  test("should show deployment logs when click on environment", async () => {
+    const steps = {
+      "git:clone": ["Cloning into 'repo'..."],
+      "state:get": ["Retrieving persisted state for environment..."],
+    };
+    mockDeploymentLogsResponses(
+      firstEnvironmentMock.latestDeploymentLog.id,
+      auth,
+      steps
+    );
+    await clickOnFirstEnvironment();
+
+    await waitFor(() => expect(outputChannelMock.show).toHaveBeenCalled());
+    await waitFor(() =>
+      expect(getOutputChannelLogs()).toEqual(
+        expect.arrayContaining([
+          "Loading logs...",
+          "$$$ git:clone",
+          "Cloning into 'repo'...",
+          "$$$ state:get",
+          "Retrieving persisted state for environment...",
+        ])
+      )
+    );
+  });
+
+  test("should clear log channel when switch between selected environments", async () => {
+    const steps = {
+      "git:clone": ["Cloning into 'repo'..."],
+      "state:get": ["Retrieving persisted state for environment..."],
+    };
+    mockDeploymentLogsResponses(
+      firstEnvironmentMock.latestDeploymentLog.id,
+      auth,
+      steps
+    );
+    await clickOnEnvironmentByName(firstEnvName);
+    await waitFor(() => expect(outputChannelMock.show).toHaveBeenCalled());
+    resetOutputChannelMocks();
+    await clickOnEnvironmentByName(secondEnvName);
+    await waitFor(() => expect(outputChannelMock.show).toHaveBeenCalled());
+    await waitFor(() => expect(outputChannelMock.clear).toHaveBeenCalled());
+  });
+
+  test("should show deployment logs when redeploy", async () => {
+    const steps = {
+      "git:clone": ["Cloning into 'repo'..."],
+      "state:get": ["Retrieving persisted state for environment..."],
+    };
+    const newDeploymentId = "new-deployment-id";
```

Comment:
this is a default value in many places across this PR. use a different value please, so we know it works as expected, and all values propagate properly

Reply:
a9304ec5ef1e551ca3f176ba7c7ba5127ad1b929